Apps/WebClient/src/ClientApp/src/views/healthInsights.vue

```diff
@@ -0,0 +1,256 @@
+import moment from "moment";
+import { Route } from "vue-router";
+import { WebClientConfiguration } from "@/models/configData";
+import {
+    ILogger,
+    IImmunizationService,
+    ILaboratoryService,
+    IMedicationService,
+} from "@/services/interfaces";
+import container from "@/plugins/inversify.config";
+import { SERVICE_IDENTIFIER } from "@/plugins/inversify";
+import { ResultType } from "@/constants/resulttype";
+import User from "@/models/user";
+import TimelineEntry from "@/models/timelineEntry";
+import MedicationTimelineEntry from "@/models/medicationTimelineEntry";
+import ImmunizationTimelineEntry from "@/models/immunizationTimelineEntry";
+import LaboratoryTimelineEntry from "@/models/laboratoryTimelineEntry";
+import MedicationStatementHistory from "../models/medicationStatementHistory";
+import RequestResult from "@/models/requestResult";
+import {
+    LaboratoryOrder,
+    LaboratoryReport,
+    LaboratoryResult,
+} from "@/models/laboratory";
+
+import TimelineLoadingComponent from "@/components/timelineLoading.vue";
+import ProtectiveWordComponent from "@/components/modal/protectiveWord.vue";
+import ErrorCardComponent from "@/components/errorCard.vue";
+
+const namespace: string = "user";
+
+@Component({
+    components: {
+        TimelineLoadingComponent,
+        ProtectiveWordComponent,
+        ErrorCard: ErrorCardComponent,
+    },
+})
+export default class HealthInsightsView extends Vue {
+    private logger: ILogger = container.get(SERVICE_IDENTIFIER.Logger);
+    @Getter("user", { namespace }) user!: User;
+    @Action("getOrders", { namespace: "laboratory" })
+    getLaboratoryOrders!: (params: {
+        hdid: string;
+    }) => Promise<RequestResult<LaboratoryOrder[]>>;
+    @Getter("webClient", { namespace: "config" })
+    config!: WebClientConfiguration;
+
+    private timelineEntries: TimelineEntry[] = [];
+    private isMedicationLoading: boolean = false;
+    private isImmunizationLoading: boolean = false;
+    private isLaboratoryLoading: boolean = false;
+    private hasErrors: boolean = false;
+    private protectiveWordAttempts: number = 0;
+
+    @Ref("protectiveWordModal")
+    readonly protectiveWordModal!: ProtectiveWordComponent;
+
+    private mounted() {
+        this.fetchMedicationStatements();
+        this.fetchImmunizations();
+        this.fetchLaboratoryResults();
+    }
+
+    private get isLoading(): boolean {
+        return (
+            this.isMedicationLoading ||
+            this.isImmunizationLoading ||
+            this.isLaboratoryLoading
+        );
+    }
+
+    private get isMedicationEnabled(): boolean {
+        return this.config.modules["Medication"];
+    }
+
+    private get isImmunizationEnabled(): boolean {
+        return this.config.modules["Immunization"];
+    }
+
+    private get isLaboratoryEnabled(): boolean {
+        return this.config.modules["Laboratory"];
+    }
+
+    private fetchMedicationStatements(protectiveWord?: string) {
```

Comment:
most of these methods are duplicated with the ones in timeline.vue. Ideally they should use the same code base, mbe a task for refactoring?

Reply:
Manuel has a task to move the fetch methods to the VUEX store!